scripts/migrate_locales.js

```diff
@@ -0,0 +1,337 @@
+  if (!object) {
+    return {}
+  }
+
+  // Key path is the key itself
+  if (object[keyPath]) {
+    const value = object[keyPath]
+    delete object[keyPath]
+    return { value, pathAsKey: true }
+  }
+
+  // Nested keys
+  const nestedKeys = keyPath.split('.')
+  const key = nestedKeys[level]
+
+  // Last key in keys path
+  if (level >= nestedKeys.length - 1) {
+    const value = object[key]
+
+    if (value) {
+      delete object[key]
+    }
+
+    return { value, pathAsKey: false }
+  }
+
+  const { value, pathAsKey } = deleteAtKey({
+    object: object[key],
+    keyPath,
+    level: level + 1,
+  })
+
+  if (object[key] && !Object.keys(object[key]).length) {
+    delete object[key]
+  }
+
+  return { value, pathAsKey }
+}
+
+function insertAtKey({ object, value, keyPath, level = 0, pathAsKey = false }) {
+  if (!object) {
+    return
+  }
+
+  // Key path is the key itself
+  if (pathAsKey) {
+    object[keyPath] = value
+    return
+  }
+
+  // Nested keys
+  const nestedKeys = keyPath.split('.')
+  const key = nestedKeys[level]
+
+  // Last key in keys path
+  if (level >= nestedKeys.length - 1) {
+    object[key] = value
+    return
+  }
+
+  if (!object[key]) {
+    object[key] = {}
+  }
+
+  insertAtKey({ object: object[key], value, keyPath, level: level + 1 })
+}
+
+/* Main functions */
+function listVersions() {
+  const versions = Object.keys(VERSIONS)
+    .sort()
+    .map((pair) => {
+      const [from, to] = pair.split('_')
+      return `* from ${from} to ${to}`
+    })
+
+  console.info(`\nSupported versions:${['', ...versions].join('\n  ')}`)
+  process.exit(0)
+}
+
+function migrate(object, dataKey) {
+  const { fromVersion, toVersion } = PARSED_ARGS
+  const changeLog = VERSIONS[[fromVersion, toVersion].join('_')]
+
+  verboseLogging(
+    `\nMigrate locale keys for ${buildColorMessage(dataKey, COLORS.BLUE)}:\n`
+  )
+
+  Object.keys(changeLog).forEach((fromKey) => {
+    const { value: possibleValue, pathAsKey } = deleteAtKey({
+      object: object[dataKey],
+      keyPath: fromKey,
+    })
+
+    if (!possibleValue) {
+      return
+    }
+
+    const toKeys = changeLog[fromKey]
+
+    verboseLogging(
+      `  - Found obsolete key ${buildColorMessage(
+        fromKey,
+        COLORS.YELLOW
+      )}, replace with:`
+    )
+    toKeys.forEach((toKey) =>
+      verboseLogging('\t*', buildColorMessage(toKey, COLORS.BLUE))
+    )
+
+    toKeys.forEach((toKey) =>
+      insertAtKey({
+        object: object[dataKey],
+        value: possibleValue,
+        keyPath: toKey,
+        pathAsKey,
+      })
+    )
+  })
+}
+
+function main() {
+  parseArgs()
+
+  const { inFile, outFile } = PARSED_ARGS
+  const inputJson = JSON.parse(fs.readFileSync(inFile))
+
+  const dataKeys = ['phrases', 'mobilePhrases']
+  dataKeys.forEach((dataKey) => migrate(inputJson, dataKey))
+  const result = JSON.stringify(inputJson, null, 2)
+
+  if (!outFile) {
+    verboseLogging('\nMigrated data:')
+    console.info(result)
```

Comment:
Is the idea here that integrator can choose to copy the output and save it somewhere else? Wonder if saving to a default file & location would be better since the output could be huge/long.

Reply:
Yes, that's the primary idea. I don't think that saving to default file & location is a good choice here because:
- Integrator can use JS objects to declare custom locale strings, while the script is emitting JSON data.
- Replacing those pieces of stuff in a JS file is complicated. I want to keep this script as simple as possible.

Reply:
Fair enough. Was only a thought anyway.